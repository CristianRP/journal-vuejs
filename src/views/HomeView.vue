<template>
  <div class="home">
    <img alt="Vue logo" src="../assets/logo.png">
  </div>
  <button class="btn btn-success" @click="goToDaybook">Daybook</button>
  <button class="btn btn-primary">Hola</button>
  <button class="btn btn-secondary">Hola</button>
</template>

<script>
export default {
  name: 'HomeView',
  components: {
  },
  methods: {
    goToDaybook() {
      this.$router.push({ name: 'no-entry' })
    }
   }
}
</script>
