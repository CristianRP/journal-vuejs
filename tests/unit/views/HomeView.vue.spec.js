import { shallowMount } from '@vue/test-utils'
import HomeView from '@/views/HomeView'

describe('Tests on HomeView component', () => {
  test('should match with the snapshot', () => {
    const wrapper = shallowMount( HomeView )
    expect( wrapper.html() ).toMatchSnapshot()
  });

  test('click on button to redired to no-entry', () => {
    const mockRouter = {
      push: jest.fn()
    }

    const wrapper = shallowMount( HomeView, {
      global: {
        mocks: {
          $router: mockRouter
        }
      }
    })

    wrapper.find( 'button' ).trigger('click')

    expect( mockRouter.push ).toHaveBeenCalled()
    expect( mockRouter.push ).toHaveBeenCalledWith({ name: 'no-entry' })
  });
});

