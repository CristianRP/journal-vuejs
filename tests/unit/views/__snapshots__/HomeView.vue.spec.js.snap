// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on HomeView component should match with the snapshot 1`] = `
<div class="home"><img alt="Vue logo" src=""></div>
<button class="btn btn-success">Daybook</button>
<button class="btn btn-primary">Hola</button>
<button class="btn btn-secondary">Hola</button>
`;
